dartagnan/src/main/java/com/dat3m/dartagnan/parsers/program/visitors/spirv/VisitorOpsMemory.java

```diff
@@ -34,23 +39,40 @@ public class VisitorOpsMemory extends SpirvBaseVisitor<Event> {
     private static final ExpressionFactory expressions = ExpressionFactory.getInstance();
     private final ProgramBuilder builder;
     private final BuiltIn builtIn;
+    private final Alignment alignment;
 
     public VisitorOpsMemory(ProgramBuilder builder) {
         this.builder = builder;
         this.builtIn = (BuiltIn) builder.getDecorationsBuilder().getDecoration(BUILT_IN);
+        this.alignment = (Alignment) builder.getDecorationsBuilder().getDecoration(ALIGNMENT);
     }
 
     @Override
     public Event visitOpStore(SpirvParser.OpStoreContext ctx) {
         Expression pointer = builder.getExpression(ctx.pointer().getText());
         Expression value = builder.getExpression(ctx.object().getText());
-        Event event = EventFactory.newStore(pointer, value);
+        List<Event> events = new ArrayList<>();
+        if (value.getType() instanceof ArrayType arrayType) {
+            for (int i = 0; i < arrayType.getNumElements(); i++) {
+                List<Expression> index = List.of(expressions.makeValue(i, types.getArchType()));
+                Expression address = expressions.makeGetElementPointer(arrayType.getElementType(), pointer, index);
+                Expression element = expressions.makeExtract(value, i);
+                Event store = EventFactory.newStore(address, element);
+                events.add(store);
+            }
```

Comment:
This only works if `element` has a basic type (e.g. int or bool). If it has an aggregate type, we should go deeper in the type hierarchy. I would start by throwing an exception in this case. The same is true for `visitOpLoad`.